File: test-suite/src/test/java/io/micronaut/docs/propagation/reactor/PropagatedContextTest.java
package io.micronaut.docs.propagation.reactor;

import io.micronaut.context.annotation.Property;
import io.micronaut.core.type.Argument;
import io.micronaut.http.HttpRequest;
import io.micronaut.http.client.HttpClient;
import io.micronaut.http.client.annotation.Client;
import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;

@Property(name = "spec.name", value = "PropagatedContextSpec")
@MicronautTest
class PropagatedContextTest {

    @Inject
    @Client("/")
    HttpClient client;

    @Test
    void testMonoRequest() {
        String hello = client.toBlocking().retrieve(HttpRequest.GET("/hello"), Argument.of(String.class));
        assertEquals("Hello, World", hello);
    }

}

File: test-suite-python/src/test/python/micronaut/docs/propagation/reactor/PropagatedContextTest.py
from typing import Annotated

import java
from jakarta.inject import Inject
from micronaut.context.annotation import Property
from micronaut.http import HttpRequest
from micronaut.http.client import HttpClient
from micronaut.http.client.annotation import Client
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

Argument = java.type("io.micronaut.core.type.Argument")
String = java.type("java.lang.String")


@Property(name="spec.name", value="PropagatedContextSpec")
@MicronautTest
class PropagatedContextTest:
    client: Annotated[HttpClient, Inject, Client("/")]

    @Test
    def testMonoRequest(self) -> None:
        hello = self.client.toBlocking().retrieve(HttpRequest.GET("/hello?name=Dean"), Argument.of(String))
        assert hello == "Hello, Dean"
